app/services/container_topology_service.rb

```diff
@@ -117,35 +117,30 @@ def build_entity_data(entity)
     data
   end
 
-  def entity_status(entity, kind)
-    case kind
-    when 'Vm', 'Host' then entity.power_state.capitalize
-    when 'ContainerNode'
-      ready_status = 'Unknown'
+  def entity_status(entity)
+    if entity.kind_of?(Host) || entity.kind_of?(Vm)
+      status = entity.power_state.capitalize
+    elsif entity.kind_of?(ContainerNode)
+      status = 'Unknown'
       entity.container_conditions.each do |condition|
         if condition.try(:name) == 'Ready' && condition.try(:status) == 'True'
-          ready_status = 'Ready'
+          status = condition.name
         else
-          ready_status = 'NotReady'
+          status = 'NotReady'
         end
       end
-      ready_status
-    when 'ContainerGroup' then entity.phase
-    when 'Container' then entity.state.capitalize
-    when 'ContainerReplicator'
-      if entity.current_replicas == entity.replicas
-        'OK'
-      else
-        'Warning'
-      end
-    when 'ContainerManager'
-      if entity.authentications.empty?
-        'Unknown'
-      else
-        entity.authentications.first.status.capitalize
-      end
-    else 'Unknown'
+    elsif entity.kind_of?(ContainerGroup)
+      status = entity.phase
+    elsif entity.kind_of?(Container)
+      status = entity.state.capitalize
+    elsif entity.kind_of?(ContainerReplicator)
+      status = (entity.current_replicas == entity.replicas) ? 'OK' : 'Warning'
+    elsif entity.kind_of?(ManageIQ::Providers::ContainerManager)
+      status = entity.authentications.empty? ? 'Unknown' : entity.authentications.first.status.capitalize
+    else
+      status = 'Unknown'
```

Comment:
Afaik there aren't many places in UI other than topology that display status in a single field (the topology in fact displays it as color and as part of the tooltip to backup accessibility issues). 
the ones I know about that have power_state are VM and Host, but for container related entities, this info is pretty much always taken from other fields, and in some cases it won't "fit" to add a status field.
For example, for replicator, the 'status' is considered 'ok' when number of requested replicas == the number of existing replicas. In the summary page, there are the requested replicas and existing replicas fields to display that info. Topology just displays that visually in another way.